logic/src/commonMain/kotlin/com/wire/kalium/logic/data/session/SessionRepository.kt

```diff
@@ -1,168 +1,153 @@
-                                userId = oldSession.userId,
-                                tokenType = accessTokenDTO.tokenType,
-                                accessToken = accessTokenDTO.value,
-                                refreshToken = refreshTokenDTO?.value ?: oldSession.refreshToken,
-                                oldSession.serverLinks,
-                                ssoId = oldSession.ssoId
-                            )
-                        )
-                    }.flatMap { userSession(userId) }
-                }
-            }
+    override suspend fun storeSession(
+        serverConfigId: String,
+        ssoId: SsoId?,
+        authTokens: AuthTokens
+    ): Either<StorageFailure, Unit> =
+        wrapStorageRequest {
+            accountsDAO.insertOrReplace(
+                idMapper.toDaoModel(authTokens.userId),
+                sessionMapper.toSsoIdEntity(ssoId),
+                serverConfigId
+            )
+        }.flatMap {
+            wrapStorageRequest { authTokenStorage.addOrReplace(sessionMapper.toAuthTokensEntity(authTokens)) }
         }
 
-    override fun allSessions(): Either<StorageFailure, List<AuthSession>> =
-        wrapStorageRequest { sessionStorage.allSessions()?.values?.toList()?.map { sessionMapper.fromPersistenceSession(it) } }
+    override suspend fun allSessions(): Either<StorageFailure, List<AccountInfo>> =
+        wrapStorageRequest { accountsDAO.allAccountList() }.map { it.map { sessionMapper.fromAccountInfoEntity(it) } }
 
-    override fun allSessionsFlow(): Flow<List<AuthSession>> =
-        sessionStorage.allSessionsFlow()
-            .map { it.values.toList().map { sessionMapper.fromPersistenceSession(it) } }
+    override suspend fun allSessionsFlow(): Flow<List<AccountInfo>> =
+        accountsDAO.observeAllAccountList()
+            .map { it.map { sessionMapper.fromAccountInfoEntity(it) } }
 
-    override fun allValidSessions(): Either<StorageFailure, List<AuthSession>> =
-        wrapStorageRequest {
-            sessionStorage.allSessions()?.filter { it.value is AuthSessionEntity.Valid }?.values?.toList()
-                ?.map { sessionMapper.fromPersistenceSession(it) }
-        }
+    override suspend fun allValidSessions(): Either<StorageFailure, List<AccountInfo.Valid>> =
+        wrapStorageRequest { accountsDAO.allValidAccountList() }
+            .map { it.map { AccountInfo.Valid(idMapper.fromDaoModel(it.userIDEntity)) } }
 
-    override fun allValidSessionsFlow(): Flow<List<AuthSession>> =
-        sessionStorage.allSessionsFlow()
-            .map {
-                it.values.filterIsInstance<AuthSessionEntity.Valid>().toList().map { sessionMapper.fromPersistenceSession(it) }
-            }
+    override suspend fun allValidSessionsFlow(): Flow<List<AccountInfo>> =
+        accountsDAO.observerValidAccountList()
+            .map { it.map { AccountInfo.Valid(idMapper.fromDaoModel(it.userIDEntity)) } }
+    //.wrapStorageRequest()
```

Comment:
do you need this comment anymore?

Reply:
not for now i will replace it with a TODO